pkg/cloud/azure/actuators/machine/actuator.go

```diff
@@ -128,72 +146,78 @@ func (a *Actuator) Create(ctx context.Context, cluster *clusterv1.Cluster, machi
+		networkInterfaceSpec.PublicLoadBalancerName = azure.DefaultPublicLBName
+		networkInterfaceSpec.InternalLoadBalancerName = azure.DefaultInternalLBName
+		networkInterfaceSpec.NatRule = 0
+	default:
+		return errors.Errorf("Unknown value %q for label `set` on machine %q, skipping machine creation", set, scope.Machine.Name)
 	}
 
-	nic, err := networkSvc.CreateDefaultVMNetworkInterface(scope.ClusterConfig.ResourceGroup, scope.Machine)
+	err = networkinterfaces.NewService(scope.Scope).CreateOrUpdate(ctx, networkInterfaceSpec)
 	if err != nil {
 		klog.Errorf("Unable to create VM network interface: %+v", err)
 		return &controllerError.RequeueAfterError{
 			RequeueAfter: time.Second * 30,
 		}
 	}
 
-	if scope.Network().APIServerLB.BackendPool.ID == "" {
-		klog.Errorf("Unable to find backend pool ID. Retrying...")
-		return &controllerError.RequeueAfterError{
-			RequeueAfter: time.Second * 15,
-		}
-	}
-
-	err = networkSvc.ReconcileNICBackendPool(*nic.Name, scope.Network().APIServerLB.BackendPool.ID)
+	decoded, err := base64.StdEncoding.DecodeString(scope.MachineConfig.SSHPublicKey)
 	if err != nil {
-		klog.Errorf("Unable to reconcile backend pool attachment: %+v", err)
-		return &controllerError.RequeueAfterError{
-			RequeueAfter: time.Second * 30,
-		}
+		errors.Wrapf(err, "failed to decode ssh public key")
 	}
 
-	pip, err := networkSvc.CreateOrUpdatePublicIPAddress(scope.ClusterConfig.ResourceGroup, networkSvc.GetPublicIPName(machine), networkSvc.GetDefaultPublicIPZone())
+	vmSpec := &virtualmachines.Spec{
+		Name:       scope.Machine.Name,
+		NICName:    networkInterfaceSpec.Name,
+		SSHKeyData: string(decoded),
+	}
+	err = virtualmachines.NewService(scope.Scope).CreateOrUpdate(context.Background(), vmSpec)
 	if err != nil {
-		klog.Errorf("Unable to create public IP: %+v", err)
+		klog.Errorf("failed to create or get machine: %+v", err)
 		return &controllerError.RequeueAfterError{
-			RequeueAfter: time.Second * 30,
+			RequeueAfter: time.Minute,
 		}
 	}
 
-	err = networkSvc.ReconcileNICPublicIP(*nic.Name, pip)
+	scriptData, err := config.GetVMStartupScript(scope, bootstrapToken)
 	if err != nil {
-		klog.Errorf("Unable to reconcile public IP attachment: %+v", err)
-		return &controllerError.RequeueAfterError{
-			RequeueAfter: time.Second * 30,
-		}
+		return errors.Wrapf(err, "failed to get vm startup script")
 	}
 
-	vm, err := computeSvc.CreateOrGetMachine(scope, bootstrapToken, kubeConfig)
+	vmExtSpec := &virtualmachineextensions.Spec{
+		Name:       "startup_script",
+		VMName:     scope.Machine.Name,
+		ScriptData: base64.StdEncoding.EncodeToString([]byte(scriptData)),
+	}
+	err = virtualmachineextensions.NewService(scope.Scope).CreateOrUpdate(context.Background(), vmExtSpec)
 	if err != nil {
 		klog.Errorf("failed to create or get machine: %+v", err)
 		return &controllerError.RequeueAfterError{
 			RequeueAfter: time.Minute,
 		}
 	}
 
-	scope.MachineStatus.VMID = &vm.ID
```

Comment:
Are we storing the machine identifiers in Status elsewhere if we remove this?

Reply:
Yep! its moved to isVMExists